src/candela/util/vega/index.js

```diff
@@ -383,60 +383,45 @@ let merge = function (defaults, options) {
   return defaults;
 };
 
-let chart = function (template, el, initialOptions, done) {
-  let that = {};
+let parseChart = function (spec, element, options) {
+  // Use element size to set size, unless size explicitly specified or
+  // element size is zero.
+  let el = d3.select(element)[0][0];
+  let sizeOptions = {};
 
-  initialOptions = initialOptions || {};
+  const size = getElementSize(el);
+  const elWidth = size.width;
+  const elHeight = size.height;
 
-  that.el = el;
-  that.options = {};
-  that.template = template;
-
-  that.update = function (newOptions) {
-    that.options = extend(that.options, newOptions);
-
-    // Use element size to set size, unless size explicitly specified or
-    // element size is zero.
-    let el = d3.select(that.el)[0][0];
-    let sizeOptions = {};
-
-    const size = getElementSize(el);
-    const elWidth = size.width;
-    const elHeight = size.height;
-
-    if (elWidth !== 0 && elHeight !== 0) {
-      if (that.options.width === undefined) {
-        sizeOptions.width = elWidth;
-      }
-      if (that.options.height === undefined) {
-        sizeOptions.height = elHeight;
-      }
+  if (elWidth !== 0 && elHeight !== 0) {
+    if (options.width === undefined) {
+      sizeOptions.width = elWidth;
+    }
+    if (options.height === undefined) {
+      sizeOptions.height = elHeight;
     }
-    let curOptions = extend(that.options, sizeOptions);
+  }
+  let curOptions = extend(options, sizeOptions);
 
-    // Options that go directly to Vega runtime
-    let vegaOptions = {
-      el: el,
-      renderer: curOptions.renderer
-    };
+  // Options that go directly to Vega runtime
+  let vegaOptions = {
+    el: el,
+    renderer: curOptions.renderer
+  };
 
-    that.spec = transform(that.template, curOptions);
+  let vegaSpec = transform(spec, curOptions);
 
-    vg.parse.spec(that.spec, function (chartObj) {
+  // Return a promise for the Vega chart object
+  return new Promise((resolve, reject) => {
```

Comment:
Since this promise never calls `reject()`, can we just omit the `reject` argument? Maybe not; the other form for `Promise()` is to take an "executor" function, and I'm not sure how the Promise object can tell the difference between the two types of argument.